wordpress-mobile/WordPress-iOS
# Hiding dashboard cards: add context menus

Related issue: #20296

Previous PRs (dependencies):

1. #20365
2. #20369

## Changes

- Add context menus with the "Hide this" button to "Drafts", "Scheduled Posts", and "Today's Stats" cards. Now every card from the new "Personalize Home Tab" screen has a "Hide this" button in the context menu. A couple more cards require special attention and will be made "personalizable" later.
- Remove unused code left after the recent dashboard cards [redesign](https://github.com/wordpress-mobile/WordPress-iOS/pull/20303) (already confirmed that these can be removed)
- Add generalized tracking for context menus and "Hide this" actions for cards (see `BlogDashboardAnalytics`)

## Screenshots

<img width="437" alt="Screenshot 2023-03-22 at 5 55 26 PM" src="https://user-images.githubusercontent.com/1567433/227047527-23be9179-14c1-4972-aeee-e86b0a923b93.png">

## Testing

*Prerequisites*
- "Personalize Home Tab" feature flag is enabled

**Test Case 1 (Draft Post)**

- Create a draft post and open dashboard
- Verify that the "Drafts" card now has a context menu
- Tap the context menu and select "Hide this"
- Verify that the card is hidden

**Test Case 2 (Schedule Post)**

- Create a scheduled post and open dashboard
- Verify that the "Schedules Posts" card now has a context menu
- Tap the context menu and select "Hide this"
- Verify that the card is hidden

**Test Case 3 (Both)**

- Create both a draft and a scheduled post
- Hide drafts
- Verify that schedules posts are still visible

**Test Case 4 (Stats)**

- Verify that the "Today's Stats" card has a context menu with the "Hide This" button
- Tap the button and verify that the card got hidden

**Test Case 5 (Stored Per-Site)**

- Create a draft for site A and hide the "Drafts" card 
- Switch to site B and create a draft 
- Verify that the "Drafts" cards is visible on Dashboard

**Test Case 6 (Regression)**

- Create a draft post
- Verify that the header and the individual posts are still tappable separately
- Verify that the context menu button is easy to tap with a finder (should preferably be testesd on the physical device)

*Other Testing Notes*:

- Test on iOS 14 to make sure context menus are displayed correctly
- Test with large dynamic type to make sure card titles don't overlap with the context menus
- An easy way to test all cards layout is by removing filter in `BlogDashboardService.parse`

## Regression Notes
1. Potential unintended areas of impact
This change affects primarily "Drafts," "Schedules Posts", and "Today's Stats" cards, but other dashboard cards were also affected. It must be verified that the context menu layout wasn't broken for other cards, such as Blaze.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

4. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:
